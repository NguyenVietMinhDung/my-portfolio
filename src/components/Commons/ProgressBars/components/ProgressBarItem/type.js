export type Props = {
  itemId: string,
  name: string,
  value: number,
};
