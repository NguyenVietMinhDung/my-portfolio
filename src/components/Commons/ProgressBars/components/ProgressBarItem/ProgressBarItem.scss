@import '../../../../../styles/abstracts/variables';
@import '../../../../../styles/abstracts/mixins';

.progress-bar-item {
	display: flex;
	flex-flow: column wrap;

	&:not(:first-child) {
		margin-top: 25px;
	}

	.header {
		display: flex;
		flex-flow: row wrap;
		justify-content: space-between;

		.name,
		.value {
			@include font($default-font-family, inherit, $black-color, 16px, 400);
		}
	}

	.bar {
		overflow: hidden;

		height: 5px;
		margin-top: 15px;

		border-radius: 5px;
		background-color: $grey-color;

		.percentage {
			height: 5px;

			border-radius: 5px;
			background-color: $violet-color;
		}
	}
}
