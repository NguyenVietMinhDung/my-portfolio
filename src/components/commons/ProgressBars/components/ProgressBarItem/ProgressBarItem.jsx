// @flow
import React, { memo, useEffect } from 'react';
import CSSModules from 'react-css-modules';
import { useCountUp } from 'react-countup';
import styles from './ProgressBarItem.scss';
import type { Props } from './type';

const getPercentage = value => `${value}%`;

const usePercentage = (id, targetValue) => {
  const { countUp, start } = useCountUp({
    end: targetValue,
    delay: 1000,
    suffix: '%',
  });
  useEffect(() => {
    document.addEventListener(`aos:in:${id}`, start);
    return () => document.removeEventListener(`aos:in:${id}`, start);
  });
  return countUp;
};

const ProgressBarItem = (props: Props) => {
  const { name, value, itemId } = props;
  const percentage = usePercentage(itemId, value);
  return (
    <li styleName="progress-bar-item">
      <div styleName="header">
        <div styleName="name">
          {name}
        </div>
        <div
          styleName="value"
          data-aos="fade-in"
          data-aos-once="true"
          data-aos-offset="40"
          data-aos-id={itemId}
        >
          {percentage}
        </div>
      </div>
      <div styleName="bar">
        <div
          styleName="percentage"
          style={{ width: getPercentage(value) }}
          data-aos="slide-right"
          data-aos-offset="0"
          data-aos-duration="2000"
          data-aos-once="true"
        />
      </div>
    </li>
  );
};

export default memo<Props>(CSSModules(ProgressBarItem, styles));
