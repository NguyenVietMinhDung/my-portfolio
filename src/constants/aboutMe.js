export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
