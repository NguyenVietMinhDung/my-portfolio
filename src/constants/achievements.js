export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
