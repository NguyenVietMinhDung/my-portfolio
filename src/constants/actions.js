export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
