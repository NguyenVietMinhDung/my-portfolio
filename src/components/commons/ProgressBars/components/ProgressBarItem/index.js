import ProgressBarItem from './ProgressBarItem';

export default ProgressBarItem;
