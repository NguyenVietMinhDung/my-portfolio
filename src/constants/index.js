import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
