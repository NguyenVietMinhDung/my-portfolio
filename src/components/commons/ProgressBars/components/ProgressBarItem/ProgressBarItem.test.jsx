import React from 'react';
import { shallow } from 'enzyme';
import { skills as items } from '../../../../../constants/aboutMe';
import ProgressBarItem from './ProgressBarItem';

const defaultProps = items[0];

describe('<ProgressBarItem />', () => {
  test('renders correctly', () => {
    const component = shallow(<ProgressBarItem {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
