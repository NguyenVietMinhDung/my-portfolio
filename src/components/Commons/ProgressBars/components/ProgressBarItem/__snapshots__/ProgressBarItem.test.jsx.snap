// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBarItem /> renders correctly 1`] = `
<li
  className="progress-bar-item"
>
  <div
    className="header"
  >
    <div
      className="name"
    >
      JavaScript, HTML, CSS
    </div>
    <div
      className="value"
      data-aos="fade-in"
      data-aos-id="skill-1"
      data-aos-offset="40"
      data-aos-once="true"
    >
      0
    </div>
  </div>
  <div
    className="bar"
  >
    <div
      className="percentage"
      data-aos="slide-right"
      data-aos-duration="2000"
      data-aos-offset="0"
      data-aos-once="true"
      style={
        Object {
          "width": "80%",
        }
      }
    />
  </div>
</li>
`;
